alacritty_terminal/src/config/bindings.rs

```diff
@@ -71,113 +71,158 @@ pub fn default_key_bindings() -> Vec<KeyBinding> {
     let mut bindings = bindings!(
         KeyBinding;
         Key::Paste; Action::Paste;
-        Key::Copy; Action::Copy;
+        Key::Copy;  Action::Copy;
         Key::L, [ctrl: true]; Action::ClearLogNotice;
         Key::L, [ctrl: true]; Action::Esc("\x0c".into());
-        Key::Home, [alt: true]; Action::Esc("\x1b[1;3H".into());
-        Key::Home, +TermMode::APP_CURSOR; Action::Esc("\x1bOH".into());
-        Key::Home, ~TermMode::APP_CURSOR; Action::Esc("\x1b[H".into());
-        Key::End, [alt: true]; Action::Esc("\x1b[1;3F".into());
-        Key::End, +TermMode::APP_CURSOR; Action::Esc("\x1bOF".into());
-        Key::End, ~TermMode::APP_CURSOR; Action::Esc("\x1b[F".into());
-        Key::PageUp, [shift: true], ~TermMode::ALT_SCREEN; Action::ScrollPageUp;
-        Key::PageUp, [shift: true], +TermMode::ALT_SCREEN; Action::Esc("\x1b[5;2~".into());
-        Key::PageUp, [ctrl: true]; Action::Esc("\x1b[5;5~".into());
-        Key::PageUp, [alt: true]; Action::Esc("\x1b[5;3~".into());
-        Key::PageUp; Action::Esc("\x1b[5~".into());
+        Key::PageUp,   [shift: true], ~TermMode::ALT_SCREEN; Action::ScrollPageUp;
+        Key::PageUp,   [shift: true], +TermMode::ALT_SCREEN; Action::Esc("\x1b[5;2~".into());
         Key::PageDown, [shift: true], ~TermMode::ALT_SCREEN; Action::ScrollPageDown;
         Key::PageDown, [shift: true], +TermMode::ALT_SCREEN; Action::Esc("\x1b[6;2~".into());
-        Key::PageDown, [ctrl: true]; Action::Esc("\x1b[6;5~".into());
-        Key::PageDown, [alt: true]; Action::Esc("\x1b[6;3~".into());
+        Key::Home,     [shift: true], ~TermMode::ALT_SCREEN; Action::ScrollToTop;
+        Key::Home,     [shift: true], +TermMode::ALT_SCREEN; Action::Esc("\x1b[1;2H".into());
+        Key::End,      [shift: true], ~TermMode::ALT_SCREEN; Action::ScrollToBottom;
+        Key::End,      [shift: true], +TermMode::ALT_SCREEN; Action::Esc("\x1b[1;2F".into());
+        Key::Home, +TermMode::APP_CURSOR; Action::Esc("\x1bOH".into());
+        Key::Home, ~TermMode::APP_CURSOR; Action::Esc("\x1b[H".into());
+        Key::End,  +TermMode::APP_CURSOR; Action::Esc("\x1bOF".into());
+        Key::End,  ~TermMode::APP_CURSOR; Action::Esc("\x1b[F".into());
+        Key::PageUp;   Action::Esc("\x1b[5~".into());
```

Comment:
Previously these bindings were nicely sorted by `key`. I think that's a good idea to keep confusion to a minimum.

Also I don't think I've stated this before, but just wanted to make sure that I think that adding these additional `Home` and `End` bindings makes a lot of sense (for jumping to start/end of scrollback).

Reply:
I too like these new bindings :tada:

Reply:
These bindings were never sorted by key. They were placed in groups by key, however I can't group them like it was before, because our tests are position dependent.

Just a reference on how they look now and before.

![Screenshot_Wed 18 Sep 2019 02:51:45 PM MSK](https://user-images.githubusercontent.com/27620401/65146374-48688d00-da24-11e9-862f-591db61c4fea.png)

Reply:
@kchibisov I'd say we should feel free to change the tests to match the format we want here. Especially if this is going to be the order they appear in our reference `alacritty.yml` file. As a user, I'd really like to see all keys together so I can think about what I want that key to do in the same place.

This is a pretty small issue tho.